### DISCUSSION TITLE
Feature request: Increase search page masonry width

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#428] [Suggestion Box]
Tumblr has a new layout for the search page which uses the rightmost column for "Related Tags" and "Related Blogs" sections instead of search content. Users have requested a function to hide this sidebar entirely and expand the masonry view of the search results across the page, similar to the old search layout.

Before/after mockup:

<img width="1292" alt="Screen Shot 2022-01-30 at 7 36 56 PM" src="https://user-images.githubusercontent.com/8336245/151735754-811e4e5d-2f58-4aef-a936-5342014d2af2.png"> | <img width="1292" alt="Screen Shot 2022-01-30 at 7 36 56 PM copy 19-50-25-327" src="https://user-images.githubusercontent.com/8336245/151736327-b4fd2b38-0e88-4419-af36-53f8e7f44a2b.png">
-- | -- 

Unfortunately, this may be quite difficult without changes from Staff's side. Tumblr's code currently checks the browser page width to decide how many columns to put the posts in, I believe, so simply removing the sidebar content leaves a blank space there instead of adding a column. This may need to be a feature request for Tumblr itself; if this is a feature you want, I would suggest it politely at https://www.tumblr.com/support. Then again, XKit Rewritten has added seemingly-not-possible features multiple times already!

**Comment by AprilSylph** [comment]
Each masonry column is its own DOM element. This wouldn't be possible without being able to hook into and modify the React code (which I would have been doing already if it were possible/if I knew how.)

**Reply by marcustyphoon** [reply]
Eh, I mean, I bet nightpool could do it.*

\**this statement exclusively based on the fact that as far as I am concerned nightpool is a wizard. Also, I read the x8 source.*

Just for the memes, I found the relevant react prop. Definitely wouldn't actually attempt this, though :P

```js
(async () => {
  const postMasonrySelector = await keyToCss('postMasonry');
  inject(async (selector) => {
    [...document.querySelectorAll(selector)].forEach(postMasonryElement => {
      const reactKey = Object.keys(postMasonryElement).find(key => key.startsWith('__reactInternalInstance') || key.startsWith('__reactFiber'));
      const fiber = postMasonryElement[reactKey];

      console.log('masonryRendersWithSidebar:', fiber.child.memoizedProps.masonryRendersWithSidebar); 
      // masonryRendersWithSidebar: true
    });
  }, [postMasonrySelector]);
})();
```
